### ISSUE TITLE
Add vector mode options to glyph filter 

### ISSUE THREAD

**Issue by lachlangrose** [pyvista/pyvista#6823]
### Describe the feature you would like to be added.

Currently when using `DataSetFilters. glyph` the glyph is aligned using "VTK_USE_VECTOR". In some cases, it is handy to be able to orient the glyph using "VTK_USE_NORMAL". 

This is possible by adding an extra argument to glyph `vector_mode` and allowing this to be either vector or normal, similar to how color_mode works.  

### Links to VTK Documentation, Examples, or Class Definitions.

Link to c++ code https://vtk.org/doc/nightly/html/classvtkGlyph3D.html#a11eb951daf212698f21fcfcf80c76a96

### Pseudocode or Screenshots

Using normal mode:
![image](https://github.com/user-attachments/assets/febb7ed4-d0d4-45ca-a90c-ad61573539f9)
Using vector mode:
![image](https://github.com/user-attachments/assets/b65f00eb-0561-4bf0-9427-922de06ed128)

**Comment by lachlangrose** [comment]
Hi, no that is not what I'm looking for although I think that is what the solution I proposed does. I misunderstood the vtk documentation.

The issue I am having is that if I create a disc geometry using a center and normal, the orientation of the disc is different to if I use the glyph filter on a pointset with the center and the normal vector as a property. 

The solid discs are the geometries, and the donuts are the glyph representation, arrows are the direction vector. 

![image](https://github.com/user-attachments/assets/d61f9ecb-6075-44db-94b6-8e1d157c074b)

```
import pyvista as pv
import numpy as np

p1 = pv.PolyData(np.array([[0, 0, 0]]))
p1["norm"] = np.array([[1, 0, 0]])
p2 = pv.PolyData(np.array([[1, 1, 1]]))
p2["norm"] = np.array([[0, 1, 0]])
p3 = pv.PolyData(np.array([[2, 2, 2]]))
p3["norm"] = np.array([[0, 0, 1]])

disk1 = pv.Disc(center=[0, 0, 0], normal=[1, 0, 0], inner=0)
disk2 = pv.Disc(center=[1, 1, 1], normal=[0, 1, 0], inner=0)
disk3 = pv.Disc(center=[2, 2, 2], normal=[0, 0, 1], inner=0)
geom = pv.Disc()
# geom = geom.rotate_y(90)
p1 = p1.glyph(
    geom=geom,
    orient="norm",
)
p2 = p2.glyph(
    geom=geom,
    orient="norm",
)
p3 = p3.glyph(
    geom=geom,
    orient="norm",
)
view = pv.Plotter()
view.add_mesh(disk1, name="disk1", color="red")
view.add_mesh(disk2, name="disk2", color="green")
view.add_mesh(disk3, name="disk3", color="blue")
view.add_mesh(p1, name="p1", color="red")
view.add_mesh(p2, name="p2", color="green")
view.add_mesh(p3, name="p3", color="blue")
view.add_arrows(np.array([[0,0,0],[1,1,1],[2,2,2]]),np.array([[1,0,0],[0,1,0],[0,0,1]]),color='black')

view.show()
```

I can fix the issue by rotating the glyph by 90 degrees in the y direction - which is fine for what I am doing - but I don't think the differences are intuitive.

**Comment by user27182** [comment]
It's possible that the underlying vtk filter assumes the input geom is initially pointing in the +x direction. So to use the filter, you also need to make sure your geom is "pointing" in the +x direction initially. 

E.g. I noticed that the `glyph` filter uses `vtkArrowSource` for the default geom. This source returns an arrow pointing in +x direction, so that's why it "points" correctly. But, the `pyvista.Disc` object "points" in the z-direction by default, which is not what you want.

So, I think your fix for initially rotating 90 degrees seems correct. 

The documentation should probably be updated with a note about this assumption for geometries pointing in the +x direction.

**Comment by lachlangrose** [comment]
@user27182 yeah, I noticed in the geometry_objects.py file the glyph is rotated by 90 degrees for some geometries.

**Comment by user27182** [comment]
Yes. Unfortunately the direction / orientation information of the Geometric Objects is lost once they're made, so you can't easily get the "normal" or "direction" of the object afterwards.

You could add a "initial_geom_direction='x'" parameter which would auto-rotate the inputs so that they point in +x instead of the direction specified by this vector.

**Comment by MatthewFlamm** [comment]
+1 for adding the starting orientation of the glyph geometry in the docs.
